pyfstat/helper_functions.py

```diff
@@ -276,29 +274,28 @@ def run_commandline(cl, raise_error=True, return_output=True):
     logger.info("Now executing: " + cl)
     if "|" in cl:
         logger.warning(
-            "Pipe ('|') found in commandline, errors may not be" " properly caught!"
+            "Pipe ('|') found in commandline, errors may not be  properly caught!"
         )
     try:
+        completed_process = subprocess.run(
+            cl,
+            check=True,
+            shell=True,
+            capture_output=True,
+            text=True,
+        )
+        if msg := completed_process.stdout:
+            logger.info(msg)
+        if msg := completed_process.stderr:
+            logger.error(msg)
         if return_output:
-            out = subprocess.check_output(
-                cl,  # what to run
-                stderr=subprocess.STDOUT,  # catch errors
-                shell=True,  # proper environment etc
-                universal_newlines=True,  # properly display linebreaks in error/output printing
-            )
-        else:
-            subprocess.check_call(cl, shell=True)
+            return getattr(completed_process, "returncode", 0)
```

Comment:
shouldn't it always be there and this suffice?
```suggestion
            return completed_process.returncode
```

Reply:
Superseded by in-person discussion: Return the whole object.